### PR TITLE
OBJLoader2 V3.0.0

### PR DESCRIPTION
**Pull request by kaisalmen**
It is finally time to remove the beta status from `OBJLoader2` V3. I changed the following things:
- `OBJLoader2`: It now extends OBJLoader2Parser: Moved common functions (parser configuration and callback) to `OBJLoader2Parser`. Here aim was to remove redundant function definitions in both the independent parser code and in `OBJLoader2`
- `OBJLoader2Parser`: All private functions are now identified by "_". Provides functions and members for extensions
- Aligned typescript definitions with latest code changes and updated code documentation where needed.
- Polished code
- Set versions to 3.0.0

Bottom line: `OBJLoader2` and `OBJLoader2Parallel` logic are so much reduced that they can be generically addressed. Hopefully, I have time to continue to work on this in the coming weeks. See **code** of working concept if you are interested: https://github.com/kaisalmen/WWOBJLoader/blob/dev/test/assetPipelineLoader/main.src.html#L104-L132

### CODE REVIEW

**Comment by Mugen87** [comment]
https://raw.githack.com/kaisalmen/three.js/OBJLoader2_V300/examples/webgl_loader_obj2.html
https://raw.githack.com/kaisalmen/three.js/OBJLoader2_V300/examples/webgl_loader_obj2_options.html

**Comment by kaisalmen** [comment]
@Mugen87 is this a direct check to see if any examples are broken?

**Comment by mrdoob** [comment]
Thanks!

**Comment by Mugen87** [comment]
> is this a direct check to see if any examples are broken?

Yep.

File: examples/jsm/loaders/obj2/worker/parallel/OBJLoader2Parser.js [hunk 1]
```diff
@@ -130,60 +165,129 @@ OBJLoader2Parser.prototype = {
...
+	 * @private
+	 *
+	 * @param {string} type The type of event
+	 * @param {string} text Textual description of the event
+	 * @param {number} numericalValue Numerical value describing the progress
+	 */
+	_onProgress: function ( type, text, numericalValue ) {
+
+		let message = text ? text : '';
+		let event = {
```
Comment (Mugen87) [inline examples/jsm/loaders/obj2/worker/parallel/OBJLoader2Parser.js]: It seems `event` is never used in `_onProgress()`. If it is not required, the object should be deleted.

Comment (kaisalmen) [reply examples/jsm/loaders/obj2/worker/parallel/OBJLoader2Parser.js]: Yes, you are right. The signature of the function should then be simplified along. Shall I submit a new PR?

Comment (kaisalmen) [reply examples/jsm/loaders/obj2/worker/parallel/OBJLoader2Parser.js]: Done, see below.

**Comment by Mugen87** [comment]
@kaisalmen It seems the following guard always evaluates to `true`.

https://github.com/mrdoob/three.js/blob/dcd1305b91b0fbab366bcfb3cffb72dce191fc07/examples/jsm/loaders/obj2/OBJLoader2Parser.js#L1015

Do you mind having a look? (I did not want to make a new issue for this minor thing so I hope it's okay to post here 😇).

**Comment by kaisalmen** [comment]
Yes, I flipped the meaning of the expression and didn't notice. Thanks. PR is here: https://github.com/mrdoob/three.js/pull/19039
